src/coprocessor/codec/mysql/decimal.rs

```diff
@@ -688,7 +688,15 @@ fn do_div_mod(
             buf[l_idx] = dcarry as u32;
         }
         idx_to = 0;
-        let mut int_word_to = word_cnt!(l_prec.wrapping_sub(l_frac_cnt) as i8, i8) - l_idx as i8;
+
+        int_cnt_to = l_prec.wrapping_sub(l_frac_cnt) as i8 - l_idx as i8 * DIGITS_PER_WORD as i8;
+
+        let mut int_word_to = if int_cnt_to < 0 {
+            int_cnt_to / DIGITS_PER_WORD as i8
```

Comment:
shall we integrate this logic to word_cnt! instead? @AndreMouche

Reply:
I think we'd better keep compatible with `TIDB` here @breeswish